src/System.Web.OData/OData/Routing/DefaultODataPathHandler.cs

```diff
@@ -112,12 +112,10 @@ private ODataPath Parse(string serviceRoot, string odataPath, IServiceProvider r
             {
                 Contract.Assert(serviceRoot != null);
 
-                serviceRootUri = new Uri(
-                    serviceRoot.EndsWith("/", StringComparison.Ordinal)
-                        ? serviceRoot
-                        : serviceRoot + "/");
+                string serviceRootStr = serviceRoot.EndsWith("/", StringComparison.Ordinal) ? serviceRoot : serviceRoot + "/";
```

Comment:
I'm not sure why. I found the same problem on Google, but no one knows how to solve it, so I tried to modify the source code and successfully implemented my OData query.